packages/cubejs-schema-compiler/src/compiler/transpilers/CubePropContextTranspiler.ts

```diff
@@ -100,6 +100,13 @@ export class CubePropContextTranspiler implements TranspilerInterface {
 
     return {
       ObjectProperty: (path) => {
+        if (path.node.key.type === 'Identifier' && path.node.key.name === 'joins' && t.isObjectExpression(path.node.value)) {
```

Comment:
I am considering the case when some defined `joins` dynamically with `asyncModule`. It will be a breaking change for him. Theoretically, it can be done this way, but I don't know if this is used in practice.